Docs on triggers and separation of concerns (#22)
Cambridge-ICCS/green-ci: README.md

  push:
    branches: [ "main" ]

  # Triggers the workflow on pushes to open pull requests
  pull_request:

  # Allows you to run this workflow manually from the Actions tab
  workflow_dispatch:
```

In the following, we restrict attention to `pull_request` triggers because it
usually makes sense to run the full workflow when merging into `main` and the
`workflow_dispatch` option already allows for finer-grained control over
specific jobs. Returning to the case of updating a Markdown file, the test suite
above will still run if a commit is pushed to an open PR doing so. To avoid this
unnecessary test run, we can specify a list of `paths` for files that would
trigger the test suite when they are run. For example, in Python we could use:
```yml
name: MyPythonTestSuite

on:
  # Triggers the workflow on pushes to open pull requests with code changes
  pull_request:
    paths:
      - '.github/workflows/test_suite_python.yml'
      - '**.py'
      - 'requirements.txt'
```
where `test_suite_python.yml` is the name of the workflow file itself. Here the
workflow will only be run if the file itself, any Python source files, or the
repository's `requirements.txt` dependencies file change.

> For compiled languages such as C, it would likely also be a good idea to include

Reviewer: ```suggestion
For compiled languages such as C, it often also is a good idea to include
```

or something like that - "would likely also be a good idea" reads very vague
